### PR TITLE
Elemental advanced filtering

### PR DESCRIPTION
**Pull request by aalves08**
Addresses Github issue: [#6268](https://github.com/rancher/dashboard/issues/6268)

### Summary
- Add advanced filtering and column toggling to `SortableTable` component
- Advanced filtering enabled for `MachineInventory` resource on Elemental
- Add ability to create an Elemental cluster from selected items on `MachineInventory` list view
- Add banner with error message when a `MachineInventorySelectorTemplate` tied to an Elemental Cluster gets deleted (displays on the Elemental cluster config component - `machine-config/machineinventoryselectortemplate`)
- Add cluster information/name to `MachineInventory` list view
- Add info icon with tooltip to cols that cannot be filtered by on advanced filtering mode
- Add the ability to group by Cluster on `MachineInventory` list view
- Hide `Manage OS Image` from dashboard view on elemental (temporary)
- Move `Manage OS Image` menu entry to `advanced` tab (temporary)
- Change app naming (visually) to `OS Management` and setting new icon
- Other small fixes and optimisations

### Screenshot/Video
Screenshot 2022-07-07 at 11 28 14

Screenshot 2022-07-07 at 11 29 41

Screenshot 2022-07-07 at 11 29 46

Screenshot 2022-07-08 at 10 22 45

Screenshot 2022-07-12 at 14 38 46

Screenshot 2022-07-18 at 14 32 26

### CODE REVIEW

**Comment by aalves08** [comment]
@richard-cox this need a rebase from https://github.com/rancher/dashboard/pull/6225 and fixing possible merge conflicts, but bulk of work is done.
I'll ping you once this PR is "truly" ready for review. Thanks

File: shell/components/SortableTable/filtering.js [hunk 1]
```diff
@@ -150,6 +177,24 @@ function columnsToSearchField(columns) {
 
 const ipLike = /^[0-9a-f\.:]+$/i;
 
+function handleStringSearch(searchFields, searchTokens, row) {
+  for ( let j = 0 ; j < searchTokens.length ; j++ ) {
```
Comment (cnotv) [inline shell/components/SortableTable/filtering.js:181]: I think you want to use [`.every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every).

```ts
return searchTokens.every(token => {
  const expect =!token.substr(0, 1) === '!'; // could you please add comments to this, why is it needed?
  const cleanToken = expect ? token.substr(1) : token;

  return cleanToken && !(matches(searchFields, cleanToken, row) !== expect)
})
```

**Review by cnotv — CHANGES_REQUESTED** [review]
There's some hints for coding, but some parts should really worked out (table, filtering). I'll take a round in the UI later on.

File: shell/components/SortableTable/filtering.js (hunk 1 above)
Comment (cnotv) [reply shell/components/SortableTable/filtering.js:181]: Also it would be great to extend it for a more generic scope, instead of just `'!'`

**Comment by cnotv** [comment]
During creation of a new Machine Registrations I get the error:
 
Screenshot 2022-08-10 at 13 51 04

Not sure why I cannot see the source code as this is supposed to have a source map for debugging and error tracing.

**Comment by cnotv** [comment]
I was not able to find `Group by` or added filters/key as in the screenshot
The count of the machine in the sidebar does not update on creation till page refresh.

Edge cases for excessive number for filters seems covered.

It would be great if filter search field was automatically focused and confirm by pressing enter. This last one can be achieved by wrapping the filter in a form and assign the submit event, then switch the button to submit.

Screenshot 2022-08-10 at 14 06 52

**Comment by nwmac** [comment]
@aalves08 Can you separate out the changes just for advanced filtering?

**Comment by aalves08** [comment]
@nwmac @richard-cox going to close this PR.

Work has been merged to dashboard with the following PRs:
https://github.com/rancher/dashboard/pull/7168
https://github.com/rancher/dashboard/pull/7134

**Pull request closed without merging**